src/reason_parser.mly

```diff
@@ -2560,6 +2563,12 @@ _expr:
       }
   | expr as_loc(infix_operator) expr
       { mkinfix $1 $2 $3 }
+  | expr VARIABLEINFIXOP expr
+      {
+        let attribute = simple_ghost_text_attr "infix" in
+        let name = mkexp ~ghost:true ~attrs:(attribute) (Pexp_ident(ghloc (Longident.parse $2))) in
```

Comment:
You need to pass location of the second argument to both mkexp here and ghloc here. Otherwise this will confuse our IDE support. 

Take a look at how mkinfix does it.